### PR TITLE
Port to asn1-combinators 0.2.0

### PR DESCRIPTION
**Pull request by emillon**
Highlights:

- mini-DSL is moved to `Asn.S`
- OID API changes a bit (`of_string` now returns an option)

### CODE REVIEW

**Comment by emillon** [comment]
I'll try to find a nice solution for the ppx_type_conv problem.

**Comment by emillon** [comment]
OK, I finally found the correct incantation: conflicting with `ppx_driver.v0.9.1`.

File: src/kp_derivable.ml [hunk 1]
```diff
@@ -71,24 +71,37 @@ end
...
 module Asn_OID = struct
   type t = Asn.OID.t
-  let show = Asn.OID.to_string
-  let pp = pp_of_to_string show
-  let compare a b =
-    String.compare (show a) (show b)
-  let equal a b = compare a b = 0
+  [@@deriving eq,ord,show]
+
+  let show oid =
```
Comment (NathanReb) [inline src/kp_derivable.ml]: Why not sticking to the derived version ?

Comment (emillon) [reply src/kp_derivable.ml]: Indeed, that works. Fixed!

**Pull request merged**